pkg/services/app/commit.go

```diff
@@ -20,19 +19,8 @@ type CommitParams struct {
 }
 
 // GetCommits gets a list of commits from the repo/branch saved in the app manifest
-func (a *App) GetCommits(params CommitParams) ([]gitprovider.Commit, error) {
-	ctx := context.Background()
-
-	app, err := a.Kube.GetApplication(ctx, types.NamespacedName{Name: params.Name, Namespace: params.Namespace})
-	if err != nil {
-		return nil, fmt.Errorf("unable to get application for %s %w", params.Name, err)
-	}
-
-	if app.Spec.SourceType == "helm" {
-		return nil, fmt.Errorf("unable to get commits for a helm chart")
-	}
-
-	owner, err := utils.GetOwnerFromUrl(app.Spec.URL)
+func (a *App) GetCommits(params CommitParams, application *wego.Application) ([]gitprovider.Commit, error) {
+	owner, err := utils.GetOwnerFromUrl(application.Spec.URL)
```

Comment:
The new `NormalizedRepoURL` type can help here. Same with the `repoName` variable below.

https://github.com/weaveworks/weave-gitops/blob/8b4fc1352305212fae9077f6ac1938bab6799d03/pkg/gitproviders/provider.go#L541